PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+        switch ownershipType {
+        case nil:
+            return .purchased
+        case "PURCHASED":
+            return .purchased
+        case "FAMILY_SHARED":
+            return .familyShared
+        default:
+            // TODO(post-migration check): Logging?
+            return .unknown
+        }
+    }
+
+    class func parsePeriodType(periodType: String?) -> PeriodType {
+        switch periodType {
+        case "normal":
+            return .normal
+        case "intro":
+            return .intro
+        case "trial":
+            return .trial
+        default:
+            // TODO(post-migration check): Also handles nil.
+            return .normal
+        }
+    }
+
+    class func parseStore(store: String?) -> Store {
+        switch store {
+        case "app_store":
+            return .appStore
+        case "mac_app_store":
+        return .macAppStore
+        case "play_store":
+            return .playStore
+        case "stripe":
+            return .stripe
+        case "promotional":
+            return .promotional
+        default:
+            // TODO(post-migration check): Logging?
+            return .unknownStore
+        }
+    }
+
+    class func willRenewWithExpirationDate(expirationDate: Date?,
+                                           store: Store,
+                                           unsubscribeDetectedAt: Date?,
+                                           billingIssueDetectedAt: Date?) -> Bool {
+        let isPromo = store == .promotional
+        let isLifetime = expirationDate == nil
+        let hasUnsubscribed = unsubscribeDetectedAt != nil
+        let hasBillingIssues = billingIssueDetectedAt != nil
+
+        return !(isPromo || isLifetime || hasUnsubscribed || hasBillingIssues)
+    }
+
+    public override var description: String {
+        var description = "<\(NSStringFromClass(type(of: self))): "
+        description += "identifier=\(self.identifier),\n"
+        description += "isActive=\(self.isActive),\n"
+        description += "willRenew=\(self.willRenew),\n"
+        description += "periodType=\(self.periodType),\n"
+        description += "latestPurchaseDate=\(String(describing: self.latestPurchaseDate)),\n"
+        description += "originalPurchaseDate=\(String(describing: self.originalPurchaseDate)),\n"
+        description += "expirationDate=\(String(describing: self.expirationDate)),\n"
+        description += "store=\(self.store),\n"
+        description += "productIdentifier=\(self.productIdentifier),\n"
+        description += "isSandbox=\(self.isSandbox),\n"
+        description += "unsubscribeDetectedAt=\(String(describing: self.unsubscribeDetectedAt)),\n"
+        description += "billingIssueDetectedAt=\(String(describing: self.billingIssueDetectedAt)),\n"
+        description += "ownershipType=\(self.ownershipType),\n"
+        description += ">"
+        return description
+    }
+
+    public override func isEqual(_ object: Any?) -> Bool {
```

Comment:
not needed for this PR, but we might as well add conformity to `Equatable` and use this method for both

Reply:
https://github.com/RevenueCat/purchases-ios/issues/600